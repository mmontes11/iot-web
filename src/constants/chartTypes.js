export const BARCHART = "BARCHART";
export const LINECHART = "LINECHART";
