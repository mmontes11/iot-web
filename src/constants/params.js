export const FIRST_PARAM = "FIRST_PARAM";
export const SECOND_PARAM = "SECOND_PARAM";
