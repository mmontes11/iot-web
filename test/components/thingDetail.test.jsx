import React from "react";
import { shallow } from "enzyme";
import ThingDetail from "components/thingDetail";
import { thing } from "../constants";

describe("components/thingDetail", () => {
  const wrapper = shallow(
    <ThingDetail thing={thing} onEventStatsClick={() => undefined} onMeasruementStatsClick={() => undefined} />,
  );
  it("renders", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
