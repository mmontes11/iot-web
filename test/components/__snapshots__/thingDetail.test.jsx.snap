// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/thingDetail renders 1`] = `
<div
  className="card"
>
  <div
    className="card-content"
  >
    <div
      className="columns"
    >
      <div
        className="column is-three-fifths has-text-centered"
      >
        <p
          className="title is-3 has-text-primary is-spaced"
        >
          foo
        </p>
        <p
          className="title is-6"
        >
          <strong>
            IP address:
          </strong>
           
          <a
            href="http://http://192.168.0.20"
            rel="noopener noreferrer"
            target="_blank"
          >
            http://192.168.0.20
          </a>
        </p>
        <p
          className="title is-6"
        >
          <strong>
            MQTT topic:
          </strong>
           
          <span
            className="has-text-info"
          >
            bar
          </span>
        </p>
        <p
          className="title is-6"
        >
          <strong>
            Last observation:
          </strong>
           
          <span
            className="has-text-info"
          >
            Mock Date
          </span>
        </p>
        <TagList
          label="Events:"
          tagStyle="is-warning"
          tags={
            Array [
              "foo",
              "bar",
            ]
          }
        />
        <TagList
          label="Measurements:"
          tagStyle="is-warning"
          tags={
            Array [
              "bar",
              "foo",
            ]
          }
        />
        <ThingActions
          onEventStatsClick={[Function]}
          thing={
            Object {
              "geometry": Object {
                "coordinates": Array [
                  -40.58457,
                  40.21339,
                ],
                "type": "Point",
              },
              "googleMapsUrl": "https://www.google.es/maps",
              "ip": "http://192.168.0.20",
              "lastObservation": "2018-08-18T10:30:04.158Z",
              "name": "foo",
              "supportedObservationTypes": Object {
                "event": Array [
                  "foo",
                  "bar",
                ],
                "measurement": Array [
                  "bar",
                  "foo",
                ],
              },
              "topic": "bar",
            }
          }
        />
      </div>
      <div
        className="column is-two-fifths"
      >
        <Connect(withProps(withScriptjs(withGoogleMap(Map))))
          marker={
            Object {
              "label": "foo",
              "point": Object {
                "lat": 40.21339,
                "lng": -40.58457,
              },
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;
